applications/FluidDynamicsApplication/custom_strategies/strategies/residualbased_predictorcorrector_velocity_bossak_scheme_turbulent.h

```diff
@@ -80,7 +80,7 @@ namespace Kratos {
     class TDenseSpace //= DenseSpace<double>
     >
     class ResidualBasedPredictorCorrectorVelocityBossakSchemeTurbulent : public ResidualBasedBossakVelocityScheme<TSparseSpace, TDenseSpace> {
-        class ElementDerivativesExtension : public DerivativesExtension
+        class ElementDerivativesExtension : public SchemeExtension
```

Comment:
i don't know what is being done here ... what is being done here? why do we need to pass such a variable to the element?

Reply:
I need the same bossak velocity scheme to be used with different variables. The current implementation had the VELOCITY and PRESSURE variables hard coded in to the scheme, and their derivatives (ACCELERATION) was hard coded as well. With this extension, it allows me to decouple the variables from the scheme and use the same bossak velocity scheme with different variables. This extension provides the variables of the derivatives for the scheme. It needs to be in the element, since I'm using same scheme with different variables in different submodel parts, and I couldn't think of any other way to do it instead of passing it through elements. :/

Reply:
hi @sunethwarna the point is that doing this comes at a memory price, which you are pushing also for "standard" elements.

as i understood from a quick look into your code, the element is not actually aware of the schemeextension.

at this point why don t you just hold in the scheme a kist of schemeextensions and loop onto that, asking for the element pointer when you need it? this would be actually faster and would have smaller memory overhead than using a pointer variable. also the element woukd not have a variable containing a pointer to the element itself...